tinygraph/tinygraph-tests.c

```diff
@@ -1124,6 +1124,239 @@ void test33(void) {
   assert(tinygraph_is_sorted_u32(a2, sizeof(a2) / sizeof(a2[0])));
 }
 
+void test34(void) {
+  tinygraph_rng_s rng = tinygraph_rng_construct();
+
+  const uint32_t n = 1000;
+
+  uint32_t * const nodes = malloc(n * sizeof(uint32_t));
+  uint16_t * const lngs = malloc(n * sizeof(uint16_t));
+  uint16_t * const lats = malloc(n * sizeof(uint16_t));
+
+  assert(nodes);
+  assert(lngs);
+  assert(lats);
+
+  tinygraph_iota_u32(nodes, n, 0);
+
+  for (uint32_t i = 0; i < n; ++i) {
+    lngs[i] = tinygraph_rng_random(rng);
+    lats[i] = tinygraph_rng_random(rng);
+  }
+
+  const bool ok = tinygraph_reorder(nodes, lngs, lats, n);
+  assert(ok);
+
+  free(nodes);
+  free(lngs);
+  free(lats);
+
+  tinygraph_rng_destruct(rng);
+}
+
+
+TINYGRAPH_WARN_UNUSED
+tinygraph_s construct_random_graph(tinygraph_rng_s rng, uint32_t nodes, uint32_t degree) {
```

Comment:
Random graph with no restriction on edges; meaning we can have edges spanning the whole node id range e.g. (0, 1000), (1, 999). The embedded graph below makes sure that edges are somewhat local meaning node ids in edges are small e.g. (0,1), (1, 4).